.github/workflows/release.yml

```diff
@@ -82,8 +82,6 @@ jobs:
             conventional-changelog-conventionalcommits@4.6.3
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      - name: Set PreRelease Version
-        run: echo "GORELEASER_CURRENT_TAG=v$(pulumictl get version --language generic)" >> $GITHUB_ENV
```

Comment:
This is _not_ the solution, but I made this change as a starting point for discussion.

What seems to be happening here is that the release job explicitly runs `make tfgen`, which produces changes to the file `provider/cmd/pulumi-resource-equinix/schema.json`.

I'm not clear why these changes happen; I can see that that file was updated in the upgrade-provider PR #139.  Do we need to run extra commands as part of the upgrade-provider action?  Is that something we _can_ do?

We could update the `.releaserc.json` so that the semantic-release action commits any changes it finds, but that feels wrong (the semantic release action is meant to only commit things that are specifically related to the version number change, not things that were missed by earlier actions).

Reply:
I didn't see this happen during the release job, although it had happened to me before locally. This has to do with the process to generate the examples. The upgrade-provider command uses `make tfgen`, so that was the only place I was able to add `generate_examples`, the problem was that `generate_examples` needs the plugin to be installed so that was creating a circular dependency between `make tfgen` `make provider` and `make generate_examples` which I solved with some workarounds. I think that under some circumstances that I am not aware of right now, that fix is ​​failing

Reply:
nvm I just saw the release actions error https://github.com/equinix/pulumi-equinix/actions/runs/10393580883/job/28782845874

Reply:
I was wrong about the task run in the release workflow, it's [`make only_tfgen`](https://github.com/equinix/pulumi-equinix/pull/152/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34R61).  When I run that task locally on the latest commit from `main`, I see changes to the schema file.  If `make tfgen` is already running during provider upgrade, does that mean we also have to run it after one or more of the other upgrade PRs to ensure that the codebase is up-to-date before release?

Reply:
In [#140](https://github.com/equinix/pulumi-equinix/pull/140) manual changes were introduced on the examples, that's why `make tfgen` in goreleaser produces that diff. So we need a couple of changes - First, fix the bug that was causing the examples to show dirty in the go.mod versions, and second, replace make tfgen with only_tfgen in goreleaser. We can also reinforce the message of avoiding manual changes in the examples

Reply:
I'm going to open a new PR with the required changes in the examples/schema. Do you want to update this PR to replace `make only_tfgen` in the .goreleaser.yml file?

Reply:
#140 hasn't been merged yet.  The uncommitted changes in the release job are unrelated to that PR